### ISSUE TITLE
Social Provider doesn't work with JDK 10

### ISSUE THREAD

**Issue by PawelJ-PL** [mohiva/play-silhouette#544]
* Silhouette Version: 5.0.5
* Operating System: Linux xxx 4.16.0-2-amd64 #1 SMP Debian 4.16.12-1 (2018-05-27) x86_64 GNU/Linux

* JDK: java version ```"10.0.1" 2018-04-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.1+10)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.1+10, mixed mode)```

* Expected Behavior: Standard flow, no exception

### Actual Behavior
When running application with JDK 10, I'm getting Null Pointer Exception when trying to use Social Provider (for example Google). Full stack trace:

```
play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[NullPointerException: null]]
	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:251)
	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:178)
	at play.core.server.AkkaHttpServer$$anonfun$1.applyOrElse(AkkaHttpServer.scala:363)
	at play.core.server.AkkaHttpServer$$anonfun$1.applyOrElse(AkkaHttpServer.scala:361)
	at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:413)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:37)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:91)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:81)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:91)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:43)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.lang.NullPointerException: null
	at org.apache.commons.lang3.SystemUtils.isJavaVersionAtLeast(SystemUtils.java:1642)
	at org.apache.commons.lang3.ClassUtils.isAssignable(ClassUtils.java:690)
	at org.apache.commons.lang3.reflect.TypeUtils.isAssignable(TypeUtils.java:375)
	at org.apache.commons.lang3.reflect.TypeUtils.isAssignable(TypeUtils.java:326)
	at org.apache.commons.lang3.reflect.TypeUtils.isInstance(TypeUtils.java:1104)
	at com.mohiva.play.silhouette.impl.providers.SocialProviderRegistry.$anonfun$getSeq$1(SocialProvider.scala:153)
	at com.mohiva.play.silhouette.impl.providers.SocialProviderRegistry.$anonfun$getSeq$1$adapted(SocialProvider.scala:153)
	at scala.collection.TraversableLike.$anonfun$filterImpl$1(TraversableLike.scala:248)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at scala.collection.TraversableLike.filterImpl(TraversableLike.scala:247)
	at scala.collection.TraversableLike.filterImpl$(TraversableLike.scala:245)
	at scala.collection.AbstractTraversable.filterImpl(Traversable.scala:104)
	at scala.collection.TraversableLike.filter(TraversableLike.scala:259)
	at scala.collection.TraversableLike.filter$(TraversableLike.scala:259)
	at scala.collection.AbstractTraversable.filter(Traversable.scala:104)
	at com.mohiva.play.silhouette.impl.providers.SocialProviderRegistry.getSeq(SocialProvider.scala:153)
	at com.mohiva.play.silhouette.impl.providers.SocialProviderRegistry.get(SocialProvider.scala:144)
	at gavi.auth.controllers.AuthenticationController.$anonfun$socialLogin$1(AuthenticationController.scala:57)
	at com.mohiva.play.silhouette.api.actions.UnsecuredActionBuilder.$anonfun$invokeBlock$6(UnsecuredAction.scala:152)
	at com.mohiva.play.silhouette.api.actions.UnsecuredRequestHandlerBuilder.$anonfun$invokeBlock$1(UnsecuredAction.scala:70)
	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:303)
	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:37)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.localPollAndExec(ForkJoinPool.java:977)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1605)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

The problem doesn't exists on JDK 8 (java version "1.8.0_161")

### Reproducible Test Case

call `socialProviderRegistry.get[SocialProvider]("something")`, where socialProviderRegistry is instance of **com.mohiva.play.silhouette.impl.providers.SocialProviderRegistry** using JDK 10

**Comment by akkie** [comment]
Play and Scala are not really compatible with Java 9+, so Silhouette can also not be compatible with Java 9+. Also the exception you mention here was thrown in Play code.

Silhouette is a Play module. So if the next Play major version will be released with Java 9+ support, then the next major Silhouette version will be compatible with this Play version.

I'll let the issue open as info for other users. But there is nothing that we actually can do to fix this.

**Comment by arnbak** [comment]
According to this thread https://discuss.lightbend.com/t/does-play-2-6-15-support-java-9-and-java-10/1203 its already working. 
I guess if we add the needed dependency or wait until 2.6.16, it should work?

**Comment by raxkin** [comment]
For me, the problem was that the internal dependence `org.apache.commons.lang3` was not up to date and don't have mapped the new java versions in the code, so can't identify your version. [Class with java version mappings](https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/JavaVersion.java)

Adding `"org.apache.commons" % "commons-lang3" % "3.8.1"` to you build.sbt will force to use the last version where it have the last java versions

**Comment by JamesSullivan** [comment]
Thanks Raxkin, your work around worked great.  Did look at trying to update Silhouette sbt version, the plugins and libraries so that it would not be necessary but after updating sbt version and the plugins it became evident there would be a bit of work with BuildSettings.scala, sbt-runner, etc. that is beyond me.

**Comment by akkie** [comment]
Fixed with #553
